src/storage/store/engine.rs

```diff
@@ -813,6 +831,118 @@ impl ShardEngine {
         Ok((account_root, events, validation_errors))
     }
 
+    /// Checks if a removed verification was set as the user's primary address and revokes it if necessary.
+    ///
+    /// When a verification is removed from a user's account, this function ensures that if the
+    /// removed address was also set as their primary address (for either Ethereum or Solana),
+    /// the primary address setting is also revoked to maintain consistency.
+    ///
+    fn check_and_revoke_primary_address(
+        &mut self,
+        fid: u64,
+        deleted_verification: &proto::VerificationAddAddressBody,
+        txn_batch: &mut RocksDbTransactionBatch,
+    ) -> Result<(), MessageValidationError> {
+        // Determine protocol from removal message
+        let protocol = match deleted_verification.protocol {
+            x if x == proto::Protocol::Ethereum as i32 => proto::Protocol::Ethereum,
+            x if x == proto::Protocol::Solana as i32 => proto::Protocol::Solana,
+            _ => return Err(MessageValidationError::AddressNotPartOfVerification),
+        };
+
+        // Parse address based on protocol
+        let parsed_address = match protocol {
+            proto::Protocol::Ethereum => {
+                Address::from_slice(&deleted_verification.address).to_checksum(None)
+            }
+            proto::Protocol::Solana => bs58::encode(&deleted_verification.address).into_string(),
+        };
```

Comment:
Don't parse addresses, just treat them as bytes

Reply:
Just like in the earlier case, the UserDataAdd message contains a checksum-encoded address. So if I don’t decode the bytes into a proper address here, I wouldn’t be comparing equivalent values.

From spec
```
enum UserDataType {
  USER_DATA_PRIMARY_ADDRESS_ETHEREUM = 7; 
  // Primary Ethereum Address
  // e.g., "0xd5596099ec95b32ddC3F22814785a253f6a09D56"

  USER_DATA_PRIMARY_ADDRESS_SOLANA = 8; 
  // Primary Solana Address
  // e.g., "4TciSRW38RGNiTSKmQamQvxUg4epWKBirBvG8LCh3ahZ"
}
```